In [3]:
# Importing Python Libraries

# This will be used to import all the required Python libraries for the Project


import pandas as pd
import matplotlib.pyplot as plt

In [4]:
# Importing the Data 

# The Iris data set has been downloaded from the internet in .csv format and saved in the current directory
# We then proceed to use the read_csv command to read the iris.csv file into our program
# A typical way of referencing the data when using pandas is "df" so here we are setting the data set variable to be "df"

df = pd.read_csv('iris.csv')

In [8]:
# Data at First Glance

# To get our first glance at the data - we use the head command - this will return the top 5 rows of the data set by default
# This confirms what our analysis has pointed to:
# 5 columnsm, 4 of which are numbers and the last column which is the class column containing the species of the flower

print(df.head())

   sepallength  sepalwidth  petallength  petalwidth        class
0          5.1         3.5          1.4         0.2  Iris-setosa
1          4.9         3.0          1.4         0.2  Iris-setosa
2          4.7         3.2          1.3         0.2  Iris-setosa
3          4.6         3.1          1.5         0.2  Iris-setosa
4          5.0         3.6          1.4         0.2  Iris-setosa


In [9]:
# Data at First Glance

# We can also look at the bottom of the data set using the tail function.
# The benefit of this is that it will confirm to us the number of rows in set

print(df.tail())

     sepallength  sepalwidth  petallength  petalwidth           class
145          6.7         3.0          5.2         2.3  Iris-virginica
146          6.3         2.5          5.0         1.9  Iris-virginica
147          6.5         3.0          5.2         2.0  Iris-virginica
148          6.2         3.4          5.4         2.3  Iris-virginica
149          5.9         3.0          5.1         1.8  Iris-virginica


As the index column goes from 0 to 149; this confirms to us that there are 150 rows in the data set.

In [10]:
# Data Types / Quality

# We want to confirm the data is as expected and that there are no gaps in it before we start to use it.
# We will start by looking if there are any null values
# Using an isnull function - I am asking it to count the number of null values in each colum

print(df.isnull().sum())

sepallength    0
sepalwidth     0
petallength    0
petalwidth     0
class          0
dtype: int64


In [11]:
# Data Types / Quality

# The info function will provide us with further understanding about the data

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   sepallength  150 non-null    float64
 1   sepalwidth   150 non-null    float64
 2   petallength  150 non-null    float64
 3   petalwidth   150 non-null    float64
 4   class        150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB
None


To explain my understanding of the output of df.info;

Line 1: <class 'pandas.core.frame.DataFrame'>
    -> Confirms to me that we are dealing with a Pandas DataFrame (as expected)

Line 2: RangeIndex: 150 entries, 0 to 149
    -> 150 different rows of data (correct - as per earlier research)
    
Lines 4 - 10:
    -> Small table clarifies what we are dealing with further:
        i.   That each of the columns have 150 non-null values
        ii.  That the data types for the length / width fields are float64
        iii. That the data type for the the class is an object



In [23]:
print(df.groupby('class')['sepallength'].describe()[['max','min']])

                 max  min
class                    
Iris-setosa      5.8  4.3
Iris-versicolor  7.0  4.9
Iris-virginica   7.9  4.9


SyntaxError: invalid syntax (Temp/ipykernel_5596/3827820173.py, line 1)